Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-password-form/ContestPasswordForm.tsx

```diff
@@ -1,92 +1,65 @@
-import React, { useCallback, useEffect, useState } from 'react';
-import isNil from 'lodash/isNil';
+import { useState } from 'react';
 
-import { useCurrentContest } from '../../../hooks/use-current-contest';
-import { usePageTitles } from '../../../hooks/use-page-titles';
+import useTheme from '../../../hooks/use-theme';
+import { useSubmitContestPasswordMutation } from '../../../redux/services/contestsService';
 import Form from '../../guidelines/forms/Form';
-import FormControl, { FormControlType, IFormControlOnChangeValueType } from '../../guidelines/forms/FormControl';
+import FormControl, { FormControlType } from '../../guidelines/forms/FormControl';
 import Heading, { HeadingType } from '../../guidelines/headings/Heading';
 
 import styles from './ContestPasswordForm.module.scss';
 
 interface IContestPasswordFormProps {
-    id: number;
-    isOfficial: boolean;
+  id: number;
+  isOfficial: boolean;
+  contestName: string;
+  onSuccess: () => void;
 }
 
-const ContestPasswordForm = ({ id, isOfficial }: IContestPasswordFormProps) => {
-    const {
-        state: {
-            contest,
-            contestPasswordError,
-        },
-        actions: { submitPassword },
-    } = useCurrentContest();
-    const [ passwordValue, setPasswordValue ] = useState<string>('');
+const ContestPasswordForm = (props: IContestPasswordFormProps) => {
+    const { id, isOfficial, contestName, onSuccess } = props;
 
-    const { actions: { setPageTitle } } = usePageTitles();
+    const { themeColors, getColorClassName } = useTheme();
 
-    const passwordFieldName = 'contestPassword';
+    const [ password, setPassword ] = useState<string>('');
+    const [ errorMessage, setErrorMessage ] = useState<string>('');
+    const [ isLoading, setIsLoading ] = useState<boolean>(false);
 
-    const handleOnSubmitPassword = useCallback(async () => {
-        await submitPassword({ id, isOfficial, password: passwordValue });
-    }, [ id, isOfficial, passwordValue, submitPassword ]);
+    const [ submitPassword ] = useSubmitContestPasswordMutation();
 
-    const handleOnChangeUpdatePassword = useCallback(
-        (value?: IFormControlOnChangeValueType) => {
-            setPasswordValue(isNil(value)
-                ? ''
-                : value.toString());
-        },
-        [ setPasswordValue ],
-    );
-
-    const renderErrorSpan = useCallback(
-        (message: string) => (
-            <span className={styles.errorMessage}>{message}</span>
-        ),
-        [],
-    );
+    const textColorClassName = getColorClassName(themeColors.textColor);
 
-    const renderErrorMessage = useCallback(
-        () => {
-            if (!isNil(contestPasswordError)) {
-                const { detail } = contestPasswordError;
-                return renderErrorSpan(detail);
-            }
+    const onPasswordSubmit = async () => {
+        setIsLoading(true);
+        const response = await submitPassword({ contestId: id.toString(), isOfficial, password });
```

Comment:
Group setters on consecutive rows

Reply:
what do you mean with that? the loader have to start before the initiation of the request, if I group the setters there would not be loading animation and everything would happen at once, which makes loader redundant, or I didnt understand that one correctly